packages/big-design/src/components/PillTabs/PillTabs.tsx

```diff
@@ -0,0 +1,137 @@
+import { CheckIcon, MoreHorizIcon } from '@bigcommerce/big-design-icons';
+import React, { createRef, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useWindowResizeListener } from '../../hooks';
+import { Button } from '../Button';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledFlexItem, StyledPillTab } from './styled';
+
+export interface PillTabItem {
+  id: string;
+  title: string;
+}
+
+export interface PillTabsProps {
+  items: PillTabItem[];
+  activePills: string[];
+  onPillClick: (itemId: string) => void;
+}
+
+export const PillTabs: React.FC<PillTabsProps> = ({ activePills, items, onPillClick }) => {
+  const parentRef = createRef<HTMLDivElement>();
+  const dropdownRef = createRef<HTMLDivElement>();
+  const [isMenuVisible, setIsMenuVisible] = useState(false);
+  const [pillsState, setPillsState] = useState(
+    items.map((item) => ({
+      isVisible: true,
+      item,
+      ref: createRef<HTMLDivElement>(),
+    })),
+  );
+
+  const hideOverflownPills = useCallback(() => {
```

Comment:
🍹 I'm probably **_way_** overthinking this, but: could we make this `hideOverflowPills` or `hideOverflow`? `Overflown` is the past participle of `Overfly`, while `Overflowed` is the past participle of `Overflow` but that `ed` feels weird to me.

Reply:
What about `hideAllThePillsThatDefinitelyDidNotFit` ?

Jokes aside, never thought about it! If grammar says `hideOverflowedPills` then so be it haha